### PR TITLE
Render union within message if used only once

### PR DESCRIPTION
**Pull request by daddykotex**
If used more than once, throw
If unused, the shape is not exported

### CODE REVIEW

**Comment by daddykotex** [comment]
I'm working a validator for `protoIndex`, I'm currently creating a new module (java only) to define that validator so that it can be pulled in via `smithy-build.json` when working with a Smithy spec

**Comment by daddykotex** [comment]
> I'm working a validator for `protoIndex`, I'm currently creating a new module (java only) to define that validator so that it can be pulled in via `smithy-build.json` when working with a Smithy spec

scratch that it should go into alloy

**Comment by daddykotex** [comment]
It's possible CI fail if alloy-core 0.1.2 is not yet available, but it should not be too long: https://repo1.maven.org/maven2/com/disneystreaming/alloy/alloy-core/0.1.2/

File: README.md [hunk 1]
```diff
@@ -1123,6 +1123,11 @@ message Testing {
 
 Smithy:
 ```kotlin
+structure Union {
```
Comment (daddykotex) [inline README.md:1126]: extra union is necessary to obtain the previous proto translation

**Review by lewisjkl — APPROVED** [review]

**Pull request merged**